resources/eks/clusters.go

```diff
@@ -0,0 +1,252 @@
+			Key:       &key,
+			Value:     &value,
+		})
+	}
+	return tValues
+}
+
+func (c *Client) transformClusterLoggingConfigurations(values *[]types.LogSetup) []*ClusterLoggingConfiguration {
+	var tValues []*ClusterLoggingConfiguration
+	for _, value := range *values {
+		for _, logType := range value.Types {
+			tValue := ClusterLoggingConfiguration{
+				AccountID: c.accountID,
+				Region:    c.region,
+				Enabled:   value.Enabled,
+				Type:      string(logType),
+			}
+			tValues = append(tValues, &tValue)
+		}
+	}
+	return tValues
+}
+
+func (c *Client) transformClusterPublicAccessCidrs(values *[]string) []*ClusterPublicAccessCidr {
+	var tValues []*ClusterPublicAccessCidr
+	for _, value := range *values {
+		tValues = append(tValues, &ClusterPublicAccessCidr{
+			AccountID: c.accountID,
+			Region:    c.region,
+			Cidr:      value,
+		})
+	}
+	return tValues
+}
+
+func (c *Client) transformClusterSecurityGroups(values *[]string) []*ClusterSecurityGroup {
+	var tValues []*ClusterSecurityGroup
+	for _, value := range *values {
+		tValues = append(tValues, &ClusterSecurityGroup{
+			AccountID:       c.accountID,
+			Region:          c.region,
+			SecurityGroupID: value,
+		})
+	}
+	return tValues
+}
+
+func (c *Client) transformClusterSubnets(values *[]string) []*ClusterSubnet {
+	var tValues []*ClusterSubnet
+	for _, value := range *values {
+		tValues = append(tValues, &ClusterSubnet{
+			AccountID: c.accountID,
+			Region:    c.region,
+			SubnetID:  value,
+		})
+	}
+	return tValues
+}
+
+func (c *Client) transformEKSClusters(values *[]types.Cluster) []*Cluster {
+	var tValues []*Cluster
+	for _, value := range *values {
+		tValue := Cluster{
+			AccountID:                c.accountID,
+			Region:                   c.region,
+			Name:                     value.Name,
+			Arn:                      value.Arn,
+			CertificateAuthorityData: value.CertificateAuthority.Data,
+			CreatedAt:                value.CreatedAt,
+			Endpoint:                 value.Endpoint,
+			PlatformVersion:          value.PlatformVersion,
+			VpcID:                    value.ResourcesVpcConfig.VpcId,
+			SecurityGroupID:          value.ResourcesVpcConfig.ClusterSecurityGroupId,
+			EndpointPrivateAccess:    value.ResourcesVpcConfig.EndpointPrivateAccess,
+			EndpointPublicAccess:     value.ResourcesVpcConfig.EndpointPublicAccess,
+			RoleArn:                  value.RoleArn,
+			Status:                   string(value.Status),
+			Version:                  value.Version,
+			Tags:                     c.transformClusterTags(&value.Tags),
+			LoggingConfigurations:    c.transformClusterLoggingConfigurations(&value.Logging.ClusterLogging),
+			PublicAccessCidrs:        c.transformClusterPublicAccessCidrs(&value.ResourcesVpcConfig.PublicAccessCidrs),
+			SecurityGroups:           c.transformClusterSecurityGroups(&value.ResourcesVpcConfig.SecurityGroupIds),
+			Subnets:                  c.transformClusterSubnets(&value.ResourcesVpcConfig.SubnetIds),
+		}
```

Comment:
I think `nil` checks are missing for `value.ResourcesVpcConfig` and `value.Logging` just like you did below.

Reply:
Thanks. @James-Quigley Thanks! I think you forgot to delete it from the above struct